OpenRA.Mods.RA/Scripting/Properties/InfiltrateProperties.cs

```diff
@@ -19,18 +22,23 @@ namespace OpenRA.Mods.RA.Scripting
 	[ScriptPropertyGroup("Ability")]
 	public class InfiltrateProperties : ScriptActorProperties, Requires<InfiltratesInfo>
 	{
-		readonly InfiltratesInfo info;
+		readonly IEnumerable<Infiltrates> infiltratesTraits;
 
 		public InfiltrateProperties(ScriptContext context, Actor self)
 			: base(context, self)
 		{
-			info = Self.Info.TraitInfo<InfiltratesInfo>();
+			infiltratesTraits = Self.TraitsImplementing<Infiltrates>();
```

Comment:
It is but `Tolist` won't have to walk the enumerable multiple times and shouldn't need to shuffle the heap around to create a contiguous chunk of data.

As with anything though real profiling should be done.